### PR TITLE
Psd 163/add multivalidation handler

### PR DESCRIPTION
**Pull request by genuinebrandonchan**
-added validation interface
-added validation implementation: 2 noop implementations for tests and a multivalidator implementation
-added validation object and logic to handler
-added tests for handler and multi validation

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -58,8 +59,9 @@ func (c *component) New(ctx context.Context, conf *config) (func(context.Context
 		LogFn:           domain.LoggerFromContext,
 		StatFn:          domain.StatFromContext,
 		AssetAttributor: a,
-		// AttributionFailureHandler field is simply a stub, it varies org to org
+		// AttributionFailureHandler and AttributedAssetValidator fields are stubs, it varies company to company
```
Comment (LorenBoren) [inline main.go]: Suggestion: I think adding information about how one would go about adding AttributionFailureHandlers and AttributedAssetValidators in the README would be useful.

Comment (genuinebrandonchan) [reply main.go]: i can write some stuff on the readme

Comment (faldridge) [inline main.go]: nit: I would word this a bit differently, maybe "...are stubs, as the actual implementations of these interfaces will likely vary greatly among different operators of this service."

File: pkg/assetvalidator/multivalidator.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/domain"
+)
+
+// MultiAttributtedAssetValidator is an implementation of AssetValidator which runs specified multiple validations
+// of type AssetValidator. In the event that a company needs different validation checks for an attributed asset, this implementation
+// will handle that
+type MultiAttributtedAssetValidator struct {
```
Comment (faldridge) [inline pkg/assetvalidator/multivalidator.go]: nit: typo here: "Atribbutted" => "Attributed". Also in the comment on line 9.

File: pkg/assetvalidator/validator_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+func TestMultiValidatorMultipleValidatorError(t *testing.T) {
+	noopValidator1 := NoopAttributtedAssetValidator{}
+	noopValidator2 := NoopErrorAttributtedAssetValidator{}
+	noopValidator3 := NoopErrorAttributtedAssetValidator{}
+	ctx := context.Background()
+	multiValidator := MultiAttributtedAssetValidator{validators: []domain.AssetValidator{&noopValidator1, &noopValidator2, &noopValidator3}}
+	attributedAsset := domain.NexposeAttributedAssetVulnerabilities{BusinessContext: domain.CloudAssetDetails{ARN: "invalid attribution"}}
+	result := multiValidator.Validate(ctx, attributedAsset)
+	assert.Equal(t, result.Error(), "\n\nthis will always throw an error\nthis will always throw an error")
+}
```
Comment (faldridge) [inline pkg/assetvalidator/validator_test.go]: 👍

File: pkg/assetvalidator/multivalidator.go (hunk 2 above)
Comment (genuinebrandonchan) [reply pkg/assetvalidator/multivalidator.go]: yea there's quite a lot of typos, let me fix them

File: main.go (hunk 1 above)
Comment (genuinebrandonchan) [reply main.go]: fixed, will change when i push this up

File: README.md [hunk 4]
```diff
@@ -105,6 +106,19 @@ Additional resources:
 * [serverfull](https://github.com/asecurityteam/serverfull)
 * [serverfull-gateway](https://github.com/asecurityteam/serverfull-gateway)
 
+<a id="markdown-validation-failure" name="validationfailurehandling"></a>
+### Validation and Failure Handling
+
+In the event that when an asset fails to attribute information from a given source,
```
Comment (faldridge) [inline README.md]: nit: "when" is not needed here.

**Pull request merged**